tests/ats/custom_resources.py

```diff
@@ -0,0 +1,20 @@
+import abc
+
+from pykube.objects import NamespacedAPIObject
+
+
+# just to serve as a common interface
+class NamespacedFluxCR(NamespacedAPIObject, abc.ABC):
+    pass
+
+
+class GitRepositoryCR(NamespacedFluxCR):
+    version = "source.toolkit.fluxcd.io/v1beta1"
+    endpoint = "gitrepositories"
+    kind = "GitRepository"
+
+
+class KustomizationCR(NamespacedFluxCR):
+    version = "kustomize.toolkit.fluxcd.io/v1beta1"
+    endpoint = "kustomizations"
+    kind = "Kustomization"
```

Comment:
Is there a way we could test `HelmReleases` as well?

Reply:
They are tested, but through `Kustomization` object, so there's no definition for `HelmRelease`. But it's a good idea to test it directly as well - TBH I'm not sure if that's possible. @ljakimczuk what do you say? Can you prepare a test for that in the flux-app-tests repo?

Reply:
Hmm, the test for it is partially ready. I wrapped it with the `Kustomization` CR, because when I was creating the repo I didn't know of what would be the desired way to execute the test cases, so I used the `GitRepository` along with the `Kustomization` to execute them. It seemed more consistent than implementing other way of distinguishing that, for example `test a` should use `Kustomization` CR, but `test b` use `HelmRelease`. This way all the test are executed by `Kustomization`, and what we actually test is up to the resources in a dir, some of them can be K8s-native ones, some may be GS-introduced ones, like apps, and some may be other Flux CRs veryfing other Flux's controllers.

This said, the `HelmRelease` could still be read from a file [here](https://github.com/giantswarm/flux-app-tests/blob/feature/init-with-data/tests/test_cases/simple-chart-release/input/overlay.yaml) and `Kustomization` can then be skipped. There is also `HelmRepository` missing in this directory, so far I've been creating it with the shell script running tests. I will update it.

Reply:
Thanks for the explanation. I think it does make sense to use `Kustomization` everywhere, and if it does create a `HelmRelease`, it might be good enough. I just wonder if there's a point in testing `HelmRelease` directly.

Reply:
Hmm I believe the end result would be the same 🤔

Reply:
I've opened a PR [here](https://github.com/giantswarm/flux-app-tests/pull/2) pushing the `HelmRepository` to the test case.

Reply:
Here's a ticket to add that in the 2nd bach of flux-app tests https://github.com/giantswarm/giantswarm/issues/19908